<launch>
    <arg name="model_name" default="roboy_xylophone_left_arm"/>
    <arg name="paused" default="false"/>
    <arg name="use_sim_time" default="false"/>
    <arg name="gui" default="false"/>
    <arg name="headless" default="true"/>
    <arg name="debug" default="false"/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="debug" value="$(arg debug)"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="paused" value="$(arg paused)"/>
        <arg name="use_sim_time" value="$(arg use_sim_time)"/>
        <arg name="headless" value="$(arg headless)"/>
        <!--<arg name="world_name" value="roboy_xylophone.world"/>-->
    </include>

    <param name="planning_plugin" type="string" value="ompl_interface/OMPLPlanner"/>


    <node name="sdf_spawner2" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
          args="-sdf -file $(find roboy_models)/$(arg model_name)/model.sdf -model $(arg model_name)"
    />

    <!--load robot description onto parameter server-->
    <include file="$(optenv CONFIG_PATH /home/stefanwalke/Documents/Roboy/ss18_xylophone_playing/src/roboy_moveit_configs)/$(arg model_name)_moveit_config/launch/planning_context.launch">
        <arg name="load_robot_description" value="true"/>
    </include>

    <!--start path planning-->
    <include file="$(optenv CONFIG_PATH /home/stefanwalke/Documents/Roboy/ss18_xylophone_playing/src/roboy_moveit_configs)/$(arg model_name)_moveit_config/launch/move_group.launch">
        <arg name="allow_trajectory_execution" value="false"/>
    </include>

    <node pkg="tf" type="static_transform_publisher" name="static_map" args="0 0 0 0 0 0 1  /world /map  100"/>

    <param name="robot_description_sdf" textfile="$(find roboy_models)/$(arg model_name)/model.sdf" />

    <!-- Load the parameters to the parameter server from the yaml file-->
    <rosparam file="$(find roboy_controller)/config/$(arg model_name).yaml" command="load"/>

    <node name="roboy_controller"
          pkg="roboy_controller"
          type="roboy"
          respawn="false"
          output="screen"
          args="$(arg model_name)"
          launch-prefix="gdb -ex run"
    />

    <node name="gesture_server" pkg="roboy_hand" type="gesture_execution_server" />
    <node name="gesture_client" pkg="roboy_hand" type="gesture_execution_client" />

</launch>